### ISSUE TITLE
Fix conch --auth

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5881]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)| @cyli reported|
|-|-|
|Trac ID|trac#5881|
|Type|defect|
|Created|2012-08-19 18:34:11Z|
|Branch|https://github.com/twisted/twisted/tree/fix-conch-auth-5881|

I broke the --auth option in [#4753](https://github.com/twisted/twisted/issues/4753). Normally it's possible to add multiple credentials checker using the strcred Options mixin. The override of addChecker in conch means that each new --auth option overrides the previous one, so only one checker can be added.

Fix this.

<details><summary>Searchable metadata</summary>

```
trac-id__5881 5881
type__defect defect
reporter__cyli cyli
priority__normal normal
milestone__ 
branch__branches_fix_conch_auth_5881 branches/fix-conch-auth-5881
branch_author__cyli cyli
status__closed closed
resolution__fixed fixed
component__conch conch
keywords__easy easy
time__1345401251000000 1345401251000000
changetime__1345417046000000 1345417046000000
version__None None
owner__cyli cyli
cc__z3p
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_3"></a>|@cyli commented|
|-|-|

(In [35290]) Branching to 'fix-conch-auth-5881'

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_4"></a>|@cyli commented|
|-|-|

(In [35293]) Only clear out existing checkers if adding checkers for the first time refs [#5881](https://github.com/twisted/twisted/issues/5881)

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_5"></a>|@cyli commented|
|-|-|

[build results](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/fix-conch-auth-5881)

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_6"></a>|@glyph set owner to @cyli|
|-|-|

Tests pass, fix looks pretty straightforward; merge away.

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_7"></a>|@cyli set status to `closed`|
|-|-|

(In [35300]) Merge fix-conch-auth-5881: allow multiple --auth parameters to be passed

Author: cyli
Reviewer: glyph
Fixes: [#5881](https://github.com/twisted/twisted/issues/5881)

Override default checkers if 1 --auth parameter is passed, but allow multiple
--auth parameters to add multiple checkers
